### PR TITLE
feat(credentials, volsync, cnpg): cleanup + tests

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**

⚒️ Fixes  # 

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning
- [x] I made sure the title starts with `feat(chart-name):`, `fix(chart-name):` or `chore(chart-name):`

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/spawner/_pvc.tpl [hunk 1]
```diff
@@ -102,26 +102,25 @@
...
-            {{- $resticrepository := printf "s3:%s/%s/%s/volsync/%s-volsync-%s" $credentials.url $credentials.bucket $.Release.Name $objectData.shortName $volsyncData.name  -}}
+            {{- $baseRepo := printf "s3:%s/%s" $credentials.url $credentials.bucket -}}
+            {{- $repoSuffix := printf "%s/volsync/%s-volsync-%s" $.Release.Name $objectData.shortName $volsyncData.name -}}
+            {{- $resticrepository := printf "%s/%s" $baseRepo $repoSuffix -}}
             {{- if $credentials.path -}}
-            {{- $resticrepository = printf "s3:%s/%s/%s/%s/volsync/%s-volsync-%s" $credentials.url $credentials.bucket ( $credentials.path | trimSuffix "/" ) $.Release.Name $objectData.shortName $volsyncData.name -}}
+              {{- $resticrepository = printf "%s/%s/%s" $baseRepo ($credentials.path | trimSuffix "/") $repoSuffix -}}
             {{- end -}}
 
-            {{- $resticpassword := ( $credentials.encrKey | default $.Release.Name ) -}}
```
Comment (stavros-k) [inline library/common/templates/spawner/_pvc.tpl:110]: We have validation in place to not allow empty encrKey

File: library/common/templates/spawner/_pvc.tpl [hunk 2]
```diff
@@ -102,26 +102,25 @@
 
             {{- $credentials := get $.Values.credentials $volsync.credentials -}}
 
-            {{- $resticrepository := printf "s3:%s/%s/%s/volsync/%s-volsync-%s" $credentials.url $credentials.bucket $.Release.Name $objectData.shortName $volsyncData.name  -}}
+            {{- $baseRepo := printf "s3:%s/%s" $credentials.url $credentials.bucket -}}
+            {{- $repoSuffix := printf "%s/volsync/%s-volsync-%s" $.Release.Name $objectData.shortName $volsyncData.name -}}
+            {{- $resticrepository := printf "%s/%s" $baseRepo $repoSuffix -}}
             {{- if $credentials.path -}}
-            {{- $resticrepository = printf "s3:%s/%s/%s/%s/volsync/%s-volsync-%s" $credentials.url $credentials.bucket ( $credentials.path | trimSuffix "/" ) $.Release.Name $objectData.shortName $volsyncData.name -}}
```
Comment (stavros-k) [inline library/common/templates/spawner/_pvc.tpl:107]: Splitted this in 2 parts, baseRepo and repoSuffix.
We inject the path in the middle if we have one

**Review by Ornias1993 — CHANGES_REQUESTED** [review]
maybe also just nuke google and azure completely while at it?

File: library/common/templates/spawner/_pvc.tpl (hunk 2 above)
Comment (Ornias1993) [reply library/common/templates/spawner/_pvc.tpl:107]: Lot cleaner indeed, I had it pretty garbage code quality so I could test if it actually worked correctly.
But this is better indeed!

File: library/common/templates/spawner/_pvc.tpl (hunk 1 above)
Comment (Ornias1993) [reply library/common/templates/spawner/_pvc.tpl:110]: I noticed after that change, lets keep it "not allowed to be empty" indeed.

File: library/common/templates/spawner/_pvc.tpl (hunk 2 above)
Comment (Ornias1993) [reply library/common/templates/spawner/_pvc.tpl:107]: I'm inclined to say we should set `.Values.credentials.something.baseRepo`
Somewhere earily in the pipeline.

I noticed most things just use the baseurl anyway

File: library/common-test/ci/credentials-values.yaml [hunk 3]
```diff
@@ -36,13 +36,11 @@ workload:
               type: http
               port: "{{ .Values.service.main.ports.main.port }}"
 
-
 credentials:
```
Comment (stavros-k) [inline library/common-test/ci/credentials-values.yaml:39]: btw those are ignored currently, as we don't have neither any volsync or a cnpg using them.

Should I add something there?

(Same for credList)

Comment (stavros-k) [reply library/common-test/ci/credentials-values.yaml:39]: On the other hand, we already have a volsync ci values file.

File: library/common/templates/spawner/_pvc.tpl (hunk 2 above)
Comment (stavros-k) [reply library/common/templates/spawner/_pvc.tpl:107]: cnpg looks like it uses 
`s3://$bucket/...`
and the url on another field

While volsync uses `s3:$url/$bucket`

Is there any other place we use the base repo?

**Comment by stavros-k** [comment]
I've gone over the changed files, I think it is fine, but a second look wouldn't be bad

File: library/common/templates/spawner/_pvc.tpl (hunk 2 above)
Comment (Ornias1993) [reply library/common/templates/spawner/_pvc.tpl:107]: ahh yes my bad.

**Pull request merged**